frontend/apps/reader/modules/readersearch.lua

```diff
@@ -100,48 +147,14 @@ function ReaderSearch:onShowFulltextSearchInput()
-                        self.last_search_text = self.input_dialog:getInputText()
-                        self.use_regex = self.check_button_regex.checked
-                        self.case_insensitive = not self.check_button_case.checked
-                        local regex_error = self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText())
-                        if self.use_regex and regex_error ~= 0 then
-                            logger.dbg("ReaderSearch: regex error", regex_error, SRELL_ERROR_CODES[regex_error])
-                            local error_message = _("Invalid regular expression")
-                            if SRELL_ERROR_CODES[regex_error] then
-                                error_message = error_message .. ":\n" .. SRELL_ERROR_CODES[regex_error]
-                            else
-                                error_message = error_message .. "."
-                            end
-                            UIManager:show(InfoMessage:new{ text = error_message })
-                        else
-                            UIManager:close(self.input_dialog)
-                            self:onShowSearchDialog(self.input_dialog:getInputText(), 1, self.use_regex, self.case_insensitive)
-                        end
-                    end,
+                        self:searchCallback(1)
+                    end
                 },
                 {
                     text = forward_text,
                     is_enter_default = true,
                     callback = function()
-                        if self.input_dialog:getInputText() == "" then return end
-                        self.last_search_text = self.input_dialog:getInputText()
-                        self.use_regex = self.check_button_regex.checked
-                        self.case_insensitive = not self.check_button_case.checked
-                        local regex_error = self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText())
-                        if self.use_regex and regex_error ~= 0 then
-                            logger.dbg("ReaderSearch: regex error", regex_error, SRELL_ERROR_CODES[regex_error])
-                            local error_message = _("Invalid regular expression")
-                            if SRELL_ERROR_CODES[regex_error] then
-                                error_message = error_message .. ":\n" .. SRELL_ERROR_CODES[regex_error]
-                            else
-                                error_message = error_message .. "."
-                            end
-                            UIManager:show(InfoMessage:new{ text = error_message })
-                        else
-                            UIManager:close(self.input_dialog)
-                            self:onShowSearchDialog(self.input_dialog:getInputText(), 0, self.use_regex, self.case_insensitive)
-                        end
+                        self:searchCallback(0)
```

Comment:
Please use 1 and -1 to make it clearer forwards vs backward (I would otherwise assume 1 means forward, which is not the case here :)